src/lib/channels.py

```diff
@@ -0,0 +1,154 @@
+    ]
+    #TODO Update once dates have been agreed
+    MAGAZINE_PROMO_RANGES = [
+        
+    ]
+
+    RADIO_DRAMA_RANGES = [
+        ("2019-04-19T00:00:00+03:00", "2019-04-24T09:00:00+03:00"),
+        ("2019-04-26T00:00:00+03:00", "2019-04-28T00:00:00+03:00"),
+        ("2019-05-03T00:00:00+03:00", "2019-05-05T00:00:00+03:00"),
+        ("2019-05-10T00:00:00+03:00", "2019-05-12T00:00:00+03:00"),
+        ("2019-05-17T00:00:00+03:00", "2019-05-19T00:00:00+03:00"),
+        ("2019-05-24T00:00:00+03:00", "2019-05-26T00:00:00+03:00"),
+        ("2019-05-31T00:00:00+03:00", "2019-06-01T00:00:00+03:00")
+    ]
+
+    RADIO_MAGAZINE_RANGES = [
+        ("2019-05-23T00:00:00+03:00", "2019-05-24T00:00:00+03:00"),
+        ("2019-05-30T00:00:00+03:00", "2019-05-31T00:00:00+03:00")
+    ]
+
+    DRAMA_S01E01_RANGES = [
+        ("2019-04-19T00:00:00+03:00", "2019-04-24T09:00:00+0300")
+    ]
+
+    DRAMA_S01E02_RANGES = [
+        ("2019-04-26T00:00:00+03:00", "2019-05-01T09:00:00+03:00")
+    ]
+
+    DRAMA_S01E03_RANGES = [
+        ("2019-05-03T00:00:00+03:00", "2019-05-08T09:00:00+03:00")
+    ]
+
+    DRAMA_S01E04_RANGES = [
+        ("2019-05-10T00:00:00+03:00", "2019-05-15T09:00:00+03:00")
+    ]
+
+    DRAMA_S01E05_RANGES = [
+        ("2019-05-17T00:00:00+03:00", "2019-05-22T09:00:00+03:00")
+    ]
+
+    DRAMA_S01E06_RANGES = [
+        ("2019-05-24T00:00:00+03:00", "2019-05-29T09:00:00+03:00")
+    ]
+
+    DRAMA_S01E07_RANGES = [
+        ("2019-05-31T00:00:00+03:00", "2019-06-01T00:00:00+03:00")
+    ]
+
+    MAGAZINE_S01E01_RANGES = [
+        ("2019-05-23T00:00:00+03:00", "2019-05-24T00:00:00+03:00")
+    ]
+
+    MAGAZINE_S01E02_RANGES = [
+        ("2019-05-30T00:00:00+03:00", "2019-05-31T00:00:00+03:00")
+    ]
+
+    CHANNEL_RANGES = {
+        DRAMA_PROMO_KEY: DRAMA_PROMO_RANGES,
+        #TODO Update and uncomment once dates have been agreed
+        #MAGAZINE_PROMO_KEY:MAGAZINE_PROMO_RANGES,
+        RADIO_DRAMA_KEY:RADIO_DRAMA_RANGES,
+        RADIO_MAGAZINE_KEY:RADIO_MAGAZINE_RANGES
+    }
+
+    SHOW_RANGES = {
+        DRAMA_S01E01_KEY:DRAMA_S01E01_RANGES,
+        DRAMA_S01E02_KEY:DRAMA_S01E02_RANGES,
+        DRAMA_S01E03_KEY:DRAMA_S01E03_RANGES,
+        DRAMA_S01E04_KEY:DRAMA_S01E04_RANGES,
+        DRAMA_S01E05_KEY:DRAMA_S01E05_RANGES,
+        DRAMA_S01E06_KEY:DRAMA_S01E06_RANGES,
+        DRAMA_S01E07_KEY:DRAMA_S01E07_RANGES,
+        MAGAZINE_S01E01_KEY:MAGAZINE_S01E01_RANGES,
+        MAGAZINE_S01E02_KEY:MAGAZINE_S01E02_RANGES
+    }
+
+    @staticmethod
+    def timestamp_is_in_ranges(timestamp, ranges):
+        for range in ranges:
+            if isoparse(range[0]) <= timestamp < isoparse(range[1]):
+                return True
+        return False
+
+    @classmethod
+    def set_channel_keys(cls, user, data, time_key):
+        for td in data:
+            timestamp = isoparse(td[time_key])
+
+            channel_dict = dict()
+            print(channel_dict)
```

Comment:
did you mean to keep this in?

Reply:
Nice catch forgot this here. Removed thanks!